BTCPay Server 1.2.0 blog post and related images (#9)
btcpayserver/blog: blog/_posts/btcpay-server-1-2-0.md

category:
  - "releases"
tags:
  - "btcpay-server"
  - "release"
  - "ui-ux"
  - "greenfield"
  - "api"
  - "update"
coverImage: "/images/BTCPayServer1.2.0.png"
---

It's been a few months since the last BTCPay Server major update and the team has been hard at work pushing through a number of efforts in that time, ranging from migrating our frontend UI framework to increasing again the Greenfield API capabilities, setting up the foundation for our plugin system and crushing numerous bugs!

As the **BTCPay Server 1.2.0** major update includes more than forty items, we won't cover them all in this release post, but as usual the high-time preference crowd can view directly the [full changelog](https://github.com/btcpayserver/btcpayserver/releases).

To update, simply navigate to your **Server Settings > Maintenance** tab and click **Update** or use the command *btcpay-update.sh* in the command line.

## **Continuation of our UI updates**

----

Since the beginning of 2021, our Design prodigies at BTCPay Server have been refactoring and updating numerous pages inside the stack to make them less confusing, more comprehensible to a broader userbase and overall, easier to use.

> 

Reviewer: We should add wallet UI signing interface here

> This release includes a number of these updated views, as well as the migration from Bootstrap3 to Bootstrap5. This framework migration enables us to create and implement more modern and user-friendly views.

Reviewer: Better to use regular terminology here. `pages` vs `views`. Tries to reduce wordiness in some sentences.
```suggestion
Since the beginning of 2021, our design contributors have improved a number of pages in the interface. This release includes a number of these updated pages and the migration from Bootstrap 3 to Bootstrap 5. This framework migration enables us to create and implement more modern-looking pages and provide a better experience.
```

> As examples, the *Crowdfund* and *Point-Of-Sale* creation pages have been updated to make it easier to navigate and more efficiently create your favorite Apps in BTCPay Server.

Reviewer: ```suggestion
The *Crowdfund* and *Point-Of-Sale* app creation pages have been updated - making navigation and creation of your favorite apps smoother.
```

![BTCPay 1.2.0 Crowdfund view](/images/BTCPay1.2.0Crowdfund.png)

> ![BTCPay 1.2.0 POS view](/images/BTCPay1.2.0POS.png)

Reviewer: Better images needed here, light theme, possibly a comparison `now and before`

> A number of UI improvements and fixes also made it into this release, namely relating to:

Reviewer: Pages vs views
```suggestion
A number of UI improvements and fixes also made it into this release, namely relating to the following pages:
```

> * The server settings view
> * The transaction signing view
> * The payment request view
> * The notifications view

Reviewer: ```suggestion
* The server settings
* The transaction signing
* The payment request
* The notifications
```

> For the full list, check out the [changelog](https://github.com/btcpayserver/btcpayserver/releases)

Reviewer: We probably should add images here.

## **BTCPay Plugins - Paving the road**

----

> 

Reviewer: Probably a good idea to once again mention the vision for the plugins, just to give more context to the reader

Author: [43f2264](https://github.com/btcpayserver/blog/pull/9/commits/43f2264cef2b7c07cd44de094f8a76dcea38a16a)

In this release, we also introduced the Greenfield API to the internal codebase.
This allows plugins to gain access to the many internal features they will require in the future.

This addition is paving the road towards having a plugin system in BTCPay that adds infinite functionality and customization on top of the core feature available.

## **Greenfield API**

----

The Greenfield is still being perfected and additions are being made to it in every udpate.

Today, here's what's **new and improved in Greenfield**:
* Add Delete User
* Add `CanModifyInvoices` permission
* Add text search terms to an invoice with `additionalSearchTerms`
* Add Get store Payment methods
* Add Generate Store OnChain Wallet
* Add StoreId to Invoice model
* Change `enabledOnly` filter to `enabled`

Some API documentation was also corrected to reflect what actually happens when using the endpoints.

## **Translations**

----

Last but not least, BTCPay Server 1.2.0 sees improvements and error corrections in multple translations for the checkout page in:
* Bulgarian
* Dutch
* Finnish
* Hebrew 
* Kazath
* Korean

> * Portuguese

Reviewer: We should try to credit the contributors who did the updates, should check transifex.

Author: I checked transifex and couldn't find anything except wouter for dutch.
The other items come from the changelog additions by @NicolasDorier

BTCPay Vault is an application which allows you to plug your hardware wallet and use it within your BTCPay Server - thanks to the [HWI](https://github.com/bitcoin-core/HWI) (Hardware Wallet Integration).
Due to changes in [HWI 2.0.1](https://github.com/bitcoin-core/HWI/releases/tag/2.0.1), BTCPay Server will now detect the version of the Vault application and notify you to update in case you're using an older version.

For the full list, check out the [changelog](https://github.com/btcpayserver/btcpayserver/releases)

## Top-up invoices 🧾

Lots of our community members use BTCPay Server to accept donations. Unlike our regular invoice, where a fixed price and rate are needed, the donations don't have that requirement.

Another interesting use-case is topping up the account. Certain merchants wanted to have the ability to accept whatever the amount is paid since they're crediting it to the customer's account. So whatever customer sends them, that's what they'll credit to their account.

*Top-up invoice* is a new type of invoice, which doesn't require an amount. This invoice is considered paid as soon as payment is detected.

![BTCPay 1.2.0 Crowdfund view](/images/topupinvoice.png)

> For now, the top-up invoice is available in the [pay button](https://docs.btcpayserver.org/FAQ/FAQ-Apps/#what-is-a-payment-button). Depending on the community feedback, we intend to gradually roll it out to other apps.

Reviewer 2: It is also available when creating invoice manually via the UI, and also in the Greenfield API if `amount` is null or undefined.

Reviewer: Done https://github.com/btcpayserver/blog/pull/9/commits/aa91931033fcdf9a21a3f9d6d627b289b381cf46
